### ISSUE TITLE
Changing AWS Launch Configuration ami result in error on first run

### ISSUE THREAD

**Issue by aldarund** [hashicorp/terraform#5265]
i have following config :

```
resource "aws_launch_configuration" "consul" {
  name_prefix          = "consul-rentapp-launch-"
  image_id             = "${var.consul_ami}"
  instance_type        = "${var.consul_instance_type}"
  key_name             = "${aws_key_pair.deployer.key_name}"
  iam_instance_profile = "${aws_iam_instance_profile.ecs.id}"
  security_groups      = ["${aws_security_group.ecs_node.id}"]
  user_data            = "${template_file.consul_server_user_data.rendered}"
  associate_public_ip_address = true
  lifecycle {
      create_before_destroy = true
  }
}
resource "aws_autoscaling_group" "consul" {
  name                 = "${var.consul_asg_name}"
  availability_zones   = ["${split(",", lookup(var.availability_zones, var.AWS_REGION))}"]
  launch_configuration = "${aws_launch_configuration.consul.name}"
  /* @todo - variablize */
  min_size             = "${var.consul_scale_min_size}"
  max_size             = "${var.consul_scale_max_size}"
  desired_capacity     = "${var.consul_scale_desired_size}"
  vpc_zone_identifier = ["${aws_subnet.main.id}"]
  health_check_type = "EC2"
  health_check_grace_period = "${var.consul_health_check_grace_period}"

  lifecycle {
      create_before_destroy = true
  }

}
```

When i changed ami and re queued plan, i got following error at first time. On second run it completed successfully.

```
Terraform v0.6.11

Setting up remote state...
Downloading remote modules...
Refreshing Terraform state prior to plan...
aws_autoscaling_group.consul: Destroying...
aws_launch_configuration.consul: Creating...
  associate_public_ip_address: "" => "1"
  ebs_block_device.#:          "" => "<computed>"
  ebs_optimized:               "" => "<computed>"
  enable_monitoring:           "" => "1"
  iam_instance_profile:        "" => "ecs-instance-profile"
  image_id:                    "" => "ami-8fcee4e5"
  instance_type:               "" => "t2.micro"
  key_name:                    "" => "deployer-key"
  name:                        "" => "<computed>"
  name_prefix:                 "" => "consul-rentapp-launch-"
  root_block_device.#:         "" => "<computed>"
  security_groups.#:           "" => "1"
  security_groups.3193336550:  "" => "sg-984d70e1"
  user_data:                   "" => "b151211e05f96e9a4770b08ccbefe69974354330"
aws_launch_configuration.consul: Creation complete
aws_autoscaling_group.consul: Modifying...
  launch_configuration: "consul-rentapp-launch-jnst4zkkwre6xatqlzcnp3i2s4" => "consul-rentapp-launch-2pko6i2b75ea3bfu7wzfom6hda"
Error applying plan:

1 error(s) occurred:

* aws_autoscaling_group.consul: Error updating Autoscaling group: ValidationError: AutoScalingGroup name not found - null
	status code: 400, request id: 7690c63c-da01-11e5-88c9-5fdd9cae3e19


```

**Comment by jen20** [comment]
Hi @aldarund! Thanks for opening this issue. Can you confirm what the value of the "${var.consul_as_name}" variable is to reproduce this? Thanks.

**Comment by aldarund** [comment]
@jen20 Here it is, didnt changed it, was using default.
```
variable "consul_asg_name" {
  default = "consul-asg"
}
```

**Comment by stack72** [comment]
Hi @aldarund 

My most sincere apologies for the issue here - I believe this is something that was solved a long time ago (your report was for 0.6.11). Currently, I am able to change the AMI id for a Launch config as follows:

```
resource "aws_launch_configuration" "ecs_cluster" {
  name_prefix                 = "cluster-conf-"
  instance_type               = "${var.instance_type}"
  image_id                    = "${var.ami_id}"
  iam_instance_profile        = "${aws_iam_instance_profile.default_ecs.id}"
  associate_public_ip_address = false

  security_groups = [
    "${aws_security_group.ecs_cluster.id}",
  ]
  key_name = "${var.key_name}"

  root_block_device {
    volume_type = "gp2"
    volume_size = "${var.root_volume_size}"
  }

  lifecycle {
    create_before_destroy = true
  }
}

resource "aws_autoscaling_group" "ecs_cluster" {
  name                 = "ecs-cluster"
  vpc_zone_identifier  = ["${var.subnet_ids}"]
  min_size             = "${var.min_size}"
  max_size             = "${var.max_size}"
  desired_capacity     = "${var.desired_capacity}"
  launch_configuration = "${aws_launch_configuration.ecs_cluster.name}"
  health_check_type    = "EC2"

  tag {
    key                 = "Name"
    value               = "ecs-cluster-instance"
    propagate_at_launch = true
  }

  tag {
    key                 = "Environment"
    value               = "${var.env_prefix}"
    propagate_at_launch = true
  }
}
```

Notice that in the LaunchConfiguration, I am using `name_prefix` - without doing this, we cannot update a launch configuration. This is an AWS action not a Terraform action

Hope this helps

Paul
